src/rulesets/oas2/functions/oas2UnusedDefinition.ts

```diff
@@ -0,0 +1,58 @@
+import { JsonPath } from '@stoplight/types';
+import { IFunction, IFunctionResult } from '../../../types';
+
+function* refIterator(obj: object, path: JsonPath): IterableIterator<[JsonPath, string]> {
```

Comment:
Note to self: I've ~~shamelessly stolen~~ been greatly inspired by @P0lip's ref-sibling's code. Not sure yet what to do with the JsonPath output. If not needed later, it might be worth considering to drop it.

Reply:
As noted by @P0lip, it should be safe to get rid of the returned `JsonPath`

Reply:
Fixed